### PR TITLE
Migrate core/renderers/common/renderer.js to goog.module syntax

### PR DESCRIPTION
**Pull request by moniika**
## The basics

- [x] I branched from `goog_module`
- [x] My pull request is against `goog_module`
- [x] My code follows the [style guide](
      https://developers.google.com/blockly/guides/modify/web/style-guide)
- [x] My code is presented in the form suggested in the [module
      conversion guide](https://github.com/google/blockly/issues/5026)
- [x] I have run `npm run test` locally already.

### Resolves

Part of #5026

### Proposed Changes

Converts `core/renderers/common/renderer.js` to `goog.module` syntax.

### CODE REVIEW

File: core/renderers/common/renderer.js [hunk 1]
```diff
@@ -206,11 +204,10 @@ Renderer.prototype.makeMarkerDrawer = function(
  * @return {!IPathObject} The renderer path object.
  * @package
  */
-Renderer.prototype.makePathObject = function(root,
-    style) {
-  return new PathObject(root, style,
+Renderer.prototype.makePathObject = function(root, style) {
+  return new PathObject(
+      root, style,
       /** @type {!ConstantProvider} */ (this.constants_));
```
Comment (cpcallen) [inline core/renderers/common/renderer.js]: Will clang-format let you remove the line break after `style,`?

File: core/renderers/common/renderer.js [hunk 2]
```diff
@@ -13,34 +13,40 @@
...
+/* eslint-disable-next-line no-unused-vars */
+const BlockSvg = goog.requireType('Blockly.BlockSvg');
+const Connection = goog.require('Blockly.Connection');
+const ConstantProvider = goog.require('Blockly.blockRendering.ConstantProvider');
+/* eslint-disable-next-line no-unused-vars */
+const Debug = goog.requireType('Blockly.blockRendering.Debug');
+const Drawer = goog.require('Blockly.blockRendering.Drawer');
+const InsertionMarkerManager = goog.require('Blockly.InsertionMarkerManager');
+const IRegistrable = goog.require('Blockly.IRegistrable');
+const IPathObject = goog.require('Blockly.blockRendering.IPathObject');
```
Comment (cpcallen) [inline core/renderers/common/renderer.js:27]: I'm _pretty_ sure `IRegisterable` and `IPathObject` should use `requireType` with suitable `eslint-disable`s.

**Review by cpcallen — APPROVED** [review]
LGTM bar two nits.

File: core/renderers/common/renderer.js (hunk 1 above)
Comment (moniika) [reply core/renderers/common/renderer.js]: Yes, it lets me change that. I've also updated another line to remove the line break.

File: core/renderers/common/renderer.js (hunk 2 above)
Comment (moniika) [reply core/renderers/common/renderer.js]: I've updated `IPathObject` and added `eslint-disable`s for both lines, but I will wait on discussion to resolve in  https://github.com/google/blockly/pull/5343 for `IRegisterable`.

Comment (moniika) [reply core/renderers/common/renderer.js]: Updated as part of rebase.

**Pull request merged**